### DISCUSSION TITLE
write_selection_file does not work properly.

### DISCUSSION THREAD

**Discussion by newtonapple** [ghostty-org/ghostty#3594] [Feature Requests, Ideas] [answered]
I'm on MacOS with default text editor TextEdit.  This is my ghostty config file:

```
copy-on-select = true
focus-follows-mouse = true

# keybinds
keybind = super+f=write_screen_file:open
keybind = super+shift+f=write_selection_file:open
```

`write_selection_file` doesn't seem to work properly.  

When I select multiple lines, the file TextEdit opens is empty. 
<img width="1325" alt="selection_txt_and____config" src="https://github.com/user-attachments/assets/11135688-1712-49cb-abf1-d28f2f1a5737" />

But when I select one word, the text in the selection.txt is the entire line. 
<img width="1303" alt="Cursor_and_selection_txt_and____config" src="https://github.com/user-attachments/assets/d053b9bd-b20b-4e73-be4c-744381a15297" />

Also tried it with `write_selection_file:paste`, the contents in the temp files were the same as the ones in screenshots.

**Comment by mitchellh** [comment, marked as answer]
Fixed by #4078
